src/rgw/rgw_client_io_decoimpl.h

```diff
@@ -0,0 +1,438 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_RGW_CLIENT_IO_DECOIMPL_H
+#define CEPH_RGW_CLIENT_IO_DECOIMPL_H
```

Comment:
since all of these classes are named Filters, maybe rename this header to `rgw_client_io_filters.h`?